specs/AppLifecycle/Restart/restartApi.markdown

```diff
@@ -0,0 +1,259 @@
+</tr>
+</tbody>
+</table>
+
+When an application calls RequestRestartNow, the mechanism for restart will be as follows:
+1. Application calls RequestRestartNow API.
+2. The API will CreateProcess with the Agent (a helper EXE in the framework package containing the API) and will then wait for an an acknowledgment (ACK) from the Agent.
+3. The Agent will get the executable path of the application and signal an ACK to the API. The Agent will then then terminate the application and then CreateProcess with the application (restarted).
+
+## Examples and Scenarios
+
+```c#
+public class App 
+{
+    // Let's assume this method is run when an app updates assets.
+    // The new assets have downloaded and installed, and now the app has to request a restart immediately
+    protected void UpdateInstallComplete()
+    {
+        // Checking if the assets have updated successfully and there are no pending updates
+        if (!CheckForUpdate())
+        {
+            AppRestartFailureReason result = AppLifecycle::RequestRestartNow("args for new instance");
+            switch (result)
+            {
+                case AppRestartFailureReason::RestartPending:
+                    // In this case the restart could not be completed because another restart is pending and we can try to restart again after a period of time
+                    RetryRestartAfterXSeconds();
+                    break;
+                case AppRestartFailureReason::Other:
+                    // In this case, a dialog box will appear informing of an error requesting restart
+                    ShowErrorMessage();
+                    break;
+            }
+        }
+    } 
+
+    // In this scenario, assume the app encounters an error during initialization.
+    // The app displays an error dialog, and after the user clicks OK on the dialog the app must restart immediately.
+    // This scenario simply displays logging for these use cases
+    protected void HandleInitializationError()
+    {
+        AppRestartFailureReason result = AppLifecycle::RequestRestartNow("args for new instance");
+        switch (result)
+        {
+            case AppRestartFailureReason::RestartPending:
+                Telemetry::WriteLine("Another restart is currently pending.");
+                break;
+            case AppRestartFailureReason::InvalidUser:
+                Telemetry::WriteLine("Current user is not signed in or not a valid user.");
+                break;
+            case AppRestartFailureReason::Other:
+                Telemetry::WriteLine("Failure restarting.");
+                break;
+        }
+    }
+}
+```
+
+# Telemetry
+
+This section explores the telemetry needs for the Restart feature in
+four dimensions – Usage, Reliability, Performance and Failure
+Investigation. The telemetry will include the metadata that are needed
+not only for measuring quality but also for helping with investigations
+of failures. Following is the event which can be used to capture
```

Comment:
Who has access to this telemetry? If the app developer cannot access it, then there's no point listing it in the API documentation. (Don't tell people about things they can't access.)

Reply:
RECOMMEND: Remove, it's inside baseball here.

Reply:
Removed